### PR TITLE
Fixed stacks redrawing

### LINKED ISSUE #183: Hiding series doesn't redraw stacked bar chart
Based on the example chart you have on your site, line-chart doesn't properly redraw the chart when you hide a series in a stacked bar chart. You can test it at http://bit.ly/1xYlUHy

If you hide a series that is between other series in the stack, the series above it keep hanging in the air. I would've assumed that after hiding a series there were no empty gaps and the Y-axis was re-scaled to match the visible data.

Here's a screenshot:

screen shot 2015-01-09 at 16 36 37

### LINKED ISSUE #283: Having issue with bars overlapping each other
I am using N3 to plot shift data as well as daily totals. the daily totals come up fine but the shifts come up as overlapping each other as seen below:

screen shot 2015-07-27 at 1 15 16 pm

If i use the running totals (average line) it seems to space them correctly as seen below:

screen shot 2015-07-27 at 1 15 04 pm

The only thing I can think of is I use the same value for X in both the totals and the shifts but I am not 100% sure why that would cause this. Any help is appreciated :)

Thanks,

Greg

### LINKED ISSUE #284: Vanishing Stacks On Long Queries
So after updating the code to the one in the plunkr @chaosmail sent, I tried doing a query of 2 months worth of data (by day) and the bars seem to get so small they become negative values. Please see the images below:

screen shot 2015-07-28 at 2 29 26 pm
screen shot 2015-07-28 at 2 29 14 pm

This is an example of it on codepen:
http://codepen.io/BinaryWasteland/pen/VLVPoW

### PR DESCRIPTION
**Pull request by chaosmail**
Hi Sebastien,

I fixed this annoying problem with redrawing stacks when some of them are not visible.

Demo: http://plnkr.co/edit/jOEeCp0O75fN9GwlMjr6?p=preview
Fixes #183
Fixes #283
Fixes #284

Best,
Christoph

### CODE REVIEW

**Comment by chaosmail** [comment]
There was also some problem with toggeling series when columns are mixed with other chart types; this has also been fixed. See issue #283

**Comment by chaosmail** [comment]
Another issue with negative column widths and hence invisible columns. See issue #284

**Comment by lorem--ipsum** [comment]
:poultry_leg:

**Comment by chaosmail** [comment]
@lorem--ipsum 
nice thanks for merging, this bug was quite annoying :)
I saw that on http://n3-charts.github.io/line-chart/#/examples the stacks redrawing is not working, but if you click on the codepen it is working fine there.
Which version of the library are we using for the examples (i thought it is using the latest version from the master)?

**Comment by lorem--ipsum** [comment]
Hmm it should be the latest of master. There might be a cache issue...

**Comment by chaosmail** [comment]
maybe yeah, I will recheck tomorrow

**Comment by chaosmail** [comment]
I also see that we are using `http://d3js.org/d3.v3.min.js` on CodePen examples - we could use the CDN here

**Comment by chaosmail** [comment]
hmm still not working on the example page :/

**Comment by lorem--ipsum** [comment]
That should work now. I released (1.1.11 peculiar-parrot). Rawgit has two different CDN, the one we're supposed to use in prod caches files like crazy. They suggest to use a tag name instead of just a branch in URLs, which we were doing until now. So I changed that too on gh-pages.

TL;DR RawGit cache was messing with us, it was our fault, I fixed the thingy.
